src/script/conversation/ConversationRepository.ts

```diff
@@ -802,7 +799,7 @@ export class ConversationRepository {
     const mapOfUserIds = conversationEntities.map(conversationEntity => conversationEntity.participating_user_ids());
     const userIds = flatten(mapOfUserIds);
     await this.userRepository.getUsersById(userIds);
-    conversationEntities.forEach(conversationEntity => this.fetchUsersAndEvents(conversationEntity));
+    await Promise.all(conversationEntities.map(conversationEntity => this.fetchUsersAndEvents(conversationEntity)));
```

Comment:
Here we need to await for all the conversation to be loaded before resolving (this was part of the bug)